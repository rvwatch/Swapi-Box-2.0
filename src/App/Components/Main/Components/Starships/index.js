import React, { Component } from 'react';
import { array, func } from 'prop-types';
import Card from '../Card';

class Starships extends Component {
  render() {
    const createCards = this.props.starships.map((ship, index) => {
      const favIds = this.props.favorites.map(fav => fav.id);
      const favStatus = favIds.includes(ship.id);
      return (
        <Card
          toggleFavorites={this.props.toggleFavorites}
          cardData={ship}
          isFav={favStatus}
          {...ship}
          key={ship.name + index}
        />
      );
    });

    return <section>{createCards}</section>;
  }
}

Starships.propTypes = {
  starships: array,
  favorites: array,
  toggleFavorites: func
};

export default Starships;
