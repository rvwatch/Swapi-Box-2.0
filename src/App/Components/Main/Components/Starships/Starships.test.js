import React from 'react';
import { shallow } from 'enzyme';
import Starships from './index';
import {
  mockCleanStarship,
  mockFavorites
} from '../../../../../apiCalls/MockedData';

describe('Starships ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Starships favorites={mockFavorites} starships={mockCleanStarship} />
    );
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
