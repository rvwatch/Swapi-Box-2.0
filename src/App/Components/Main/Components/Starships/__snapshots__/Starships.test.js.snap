// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starships  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Starships
    favorites={
        Array [
            Object {
              "homeworld": "Tatooine",
              "id": 1,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 2,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 3,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
          ]
    }
    starships={
        Array [
            Object {
              "crew": "Crew: 279144",
              "id": "Executor0",
              "length": "Length: 19000m/s",
              "model": "Model: Executor-class star dreadnought",
              "name": "Name: Executor",
              "passengers": "Passengers: 38000",
              "starship_class": "Class: Star dreadnought",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardData={
                    Object {
                              "crew": "Crew: 279144",
                              "id": "Executor0",
                              "length": "Length: 19000m/s",
                              "model": "Model: Executor-class star dreadnought",
                              "name": "Name: Executor",
                              "passengers": "Passengers: 38000",
                              "starship_class": "Class: Star dreadnought",
                            }
          }
          crew="Crew: 279144"
          id="Executor0"
          isFav={false}
          length="Length: 19000m/s"
          model="Model: Executor-class star dreadnought"
          name="Name: Executor"
          passengers="Passengers: 38000"
          starship_class="Class: Star dreadnought"
          toggleFavorites={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Name: Executor0",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "crew": "Crew: 279144",
            "id": "Executor0",
            "length": "Length: 19000m/s",
            "model": "Model: Executor-class star dreadnought",
            "name": "Name: Executor",
            "passengers": "Passengers: 38000",
            "starship_class": "Class: Star dreadnought",
          },
          "crew": "Crew: 279144",
          "id": "Executor0",
          "isFav": false,
          "length": "Length: 19000m/s",
          "model": "Model: Executor-class star dreadnought",
          "name": "Name: Executor",
          "passengers": "Passengers: 38000",
          "starship_class": "Class: Star dreadnought",
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
                        Object {
                                    "crew": "Crew: 279144",
                                    "id": "Executor0",
                                    "length": "Length: 19000m/s",
                                    "model": "Model: Executor-class star dreadnought",
                                    "name": "Name: Executor",
                                    "passengers": "Passengers: 38000",
                                    "starship_class": "Class: Star dreadnought",
                                  }
            }
            crew="Crew: 279144"
            id="Executor0"
            isFav={false}
            length="Length: 19000m/s"
            model="Model: Executor-class star dreadnought"
            name="Name: Executor"
            passengers="Passengers: 38000"
            starship_class="Class: Star dreadnought"
            toggleFavorites={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Name: Executor0",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "crew": "Crew: 279144",
              "id": "Executor0",
              "length": "Length: 19000m/s",
              "model": "Model: Executor-class star dreadnought",
              "name": "Name: Executor",
              "passengers": "Passengers: 38000",
              "starship_class": "Class: Star dreadnought",
            },
            "crew": "Crew: 279144",
            "id": "Executor0",
            "isFav": false,
            "length": "Length: 19000m/s",
            "model": "Model: Executor-class star dreadnought",
            "name": "Name: Executor",
            "passengers": "Passengers: 38000",
            "starship_class": "Class: Star dreadnought",
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
